packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,186 @@
+    pushEvent: {},
+    popEvent: {},
+  };
+
+  public static readonly LINK_QUEUE_EVENT_NAME = '<Link queue occupancy>';
+  public static readonly LINK_QUEUE_DIDNT_START_EMPTY_MESSAGE = 'the link queue didn\'t start empty';
+
+  /**
+   *
+   * @param {ILinkQueue} linkQueue - The link queue
+   * @param {Algebra.Operation} query - The current query
+   * @param {Logger} logger - The logger where the events are output
+   */
+  public constructor(linkQueue: ILinkQueue, query: string, logger: Logger) {
+    super(linkQueue);
+    this.logger = logger;
+
+    if (!linkQueue.isEmpty()) {
+      this.logger.warn(
+        LinkQueueLogger.LINK_QUEUE_EVENT_NAME,
+        {
+          message: LinkQueueLogger.LINK_QUEUE_DIDNT_START_EMPTY_MESSAGE,
+        },
+      );
+    }
+    this.query = query;
+
+    Object.freeze(this.query);
+  }
+
+  /**
+   * Helper function to get the reachability criteria of a link
+   * @param {ILink} link - Current link
+   * @returns {IProduceByActor | null } The reachability criteria with extra information about it if available
+   */
+  private static getActorProductorInformation(link: ILink): IProduceByActor | null {
+    const metadata = link.metadata;
+    if (metadata !== undefined && metadata[PRODUCED_BY_ACTOR] !== undefined) {
+      const { name, ...rest } = metadata[PRODUCED_BY_ACTOR];
+
+      if (name === undefined) {
+        return null;
+      }
+      return {
+        name,
+        metadata: Object.keys(rest).length === 0 ? undefined : rest,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Update the link production ratio
+   * @param {IURLStatistic} link - current link
+   * @param {keyof ILinkProductionActorRatio} event - link queue event
+   */
+  private updateLinkProductionRatio(link: IURLStatistic, event: keyof ILinkProductionActorRatio): void {
+    if (link.producedByActor !== null &&
+      this.linkProductionRatio[event][link.producedByActor?.name] !== undefined) {
+      this.linkProductionRatio[event][link.producedByActor?.name] += 1;
+    } else if (link.producedByActor !== null &&
+      this.linkProductionRatio[event][link.producedByActor?.name] === undefined) {
+      this.linkProductionRatio[event][link.producedByActor?.name] = 1;
+    } else if (this.linkProductionRatio[event].unknown === undefined) {
+      this.linkProductionRatio[event].unknown = 1;
+    } else {
+      this.linkProductionRatio[event].unknown += 1;
+    }
+  }
+
+  /**
+   * Create the information of a link queue event
+   * @param {ILink} link - current link
+   * @param {EventType} eventType - the type of event
+   * @param {ILink|undefined} parent - the parent of the link
+   * @returns {ILinkQueueEvent} current event of the link queue
+   */
+  private createLinkQueueEvent(link: ILink, eventType: EventType, parent?: ILink): ILinkQueueEvent {
+    const linkInfo: IURLStatistic = {
+      url: link.url,
+      producedByActor: LinkQueueLogger.getActorProductorInformation(link),
+      timestamp: Date.now(),
```

Comment:
Are you sure `performance.now()` doesn't work? Because it has been in Node since version 8 (2017).

Reply:
It work it's just the mocking the doesn't because `Cannot assign to read only property` on node18. But I will investigate more, if I can find a workaround.